### ISSUE TITLE
java.lang.RuntimeException: Problems reading data from Binary store 

### ISSUE THREAD

**Issue by nskvortsov** [gradle/gradle#13639]
Attempt to open a project in IntelliJ IDEA results in falure, caused by issue with binary store. This is a stable reproducer for one more variant of #8489

### Expected Behavior
Sync finishes successfully

### Current Behavior
during sync, an attempt to resolve `:grails-datastore-gorm-hibernate5:testRuntimeClasspath` failes due to serialization issue

### Steps to Reproduce 

* clone https://github.com/grails/gorm-hibernate5
* use "open or import" in IntelliJ IDEA 2020.1 or 2020.2 EAP

### Your Environment

Build scan URL: https://scans.gradle.com/s/h4ckviofqhyy4

**Comment by ljacomet** [comment]
Gradle 6.5.1 was released today with a number of fixes in that area.
Could you try it out and let us know if it resolves the issue?

**Comment by ant-druha** [comment]
Same error with 6.5.1 unfortunately.

**Comment by ljacomet** [comment]
Thanks for checking. The reproducer will help a lot for the investigation.

**Comment by ljacomet** [comment]
The underlying issue has been identified and a fix will be published with Gradle 6.6

**Comment by klpitshah** [comment]
i am still facing this with gradle 6.9. although, I'm facing this issue only in my local system. It's working fine in was. 
some observations I've made : 
As soon as i trigger gradle refresh from intellij, 2 bins get created in `/private/var/folders`. after some time both of them get deleted and the moment they get deleted, i see this build error that
`Problems reading data from Binary store in /private/var/folders/5z/t4r01fl1073_1lmpjnkdty8w0000gn/T/gradle5813954742885277157.bin offset 652767 exists? true`

I've already tried disabling parallel run for gradle, but that hasn't helped either.
